### PR TITLE
add third reality temperature and humidity sensor

### PR DESCRIPTION
**Pull request by weihuan1111**

### CODE REVIEW

**Review by weihuan1111 — COMMENTED** [review]
update

**Comment by weihuan1111** [comment]
@Koenkk please help to review the merge request, thanks

File: devices/third_reality.js [hunk 1]
```diff
@@ -136,4 +136,17 @@ module.exports = [
...
+    {
+        zigbeeModel: ['3RTHS24BZ'],
+        model: '3RTHS24BZ',
+        vendor: 'Third Reality',
+        description: 'Temperature and humidity sensor',
+        fromZigbee: [fz.battery, fz.temperature, fz.humidity],
+        toZigbee: [],
+        exposes: [e.battery(), e.temperature(), e.humidity(), e.battery_voltage()],
+        configure: async (device, coordinatorEndpoint, logger) => {
+            device.powerSource = 'Battery';
```
Comment (danieledwardgeorgehitchcock) [inline devices/third_reality.js:148]: Could you post up the database.db entry for this device? I'd be very surprised if it doesn't report it's power source on interview.. If it does, you wont need this entry

**Comment by Koenkk** [comment]
Looks good, please check the comment from @danieledwardgeorgehitchcock

File: devices/third_reality.js (hunk 1 above)
Comment (weihuan1111) [reply devices/third_reality.js:148]: It will report the power.

**Comment by weihuan1111** [comment]
> Looks good, please check the comment from @danieledwardgeorgehitchcock

The T&H will report the power.

File: devices/third_reality.js (hunk 1 above)
Comment (Koenkk) [reply devices/third_reality.js:148]: he is talking about the `powerSource`. When you force remove the device, restart z2m and pair the device; what power source shows up in the z2m frontend?

**Comment by Koenkk** [comment]
Note to self; after this merge https://github.com/Koenkk/zigbee2mqtt.io/pull/1615

File: devices/third_reality.js (hunk 1 above)
Comment (Koenkk) [reply devices/third_reality.js:148]: Can you still check this @weihuan1111 ?

Comment (weihuan1111) [reply devices/third_reality.js:148]: image

Comment (weihuan1111) [reply devices/third_reality.js:148]: 4293c425ca1f7229fe0723fe982e658

Comment (weihuan1111) [reply devices/third_reality.js:148]: dcfdc5968f1cd6f815e3e5366cb7b3e

Comment (weihuan1111) [reply devices/third_reality.js:148]: @Koenkk You can see the above picture. I have tested it and it will display information such as power

Comment (danieledwardgeorgehitchcock) [reply devices/third_reality.js:148]: Could you post up your database.db entry as requested?

Comment (weihuan1111) [reply devices/third_reality.js:148]: database.db? How to post up database.db?

Comment (danieledwardgeorgehitchcock) [reply devices/third_reality.js:148]: You'll find it in the config folder where you installed Z2M

Comment (weihuan1111) [reply devices/third_reality.js:148]: b63385ef7b53941731f0d75e43cea5a

Comment (weihuan1111) [reply devices/third_reality.js:148]: {"id":4,"type":"EndDevice","ieeeAddr":"0xa4c13847b1bede6c","nwkAddr":12222,"manufId":4659,"manufName":"Third Reality, Inc","powerSource":"Battery","modelId":"3RTHS24BZ","epList":[1],"endpoints":{"1":{"profId":260,"epId":1,"devId":770,"inClusterList":[0,1,1026,1029],"outClusterList":[25],"clusters":{"msTemperatureMeasurement":{"attributes":{"measuredValue":2800}},"msRelativeHumidity":{"attributes":{"measuredValue":6500}},"genBasic":{"attributes":{"manufacturerName":"Third Reality, Inc","powerSource":1,"zclVersion":3,"swBuildId":"1.00.14","appVersion":14,"stackVersion":2,"hwVersion":0}},"genPowerCfg":{"attributes":{"batteryVoltage":30,"batteryPercentageRemaining":200}}},"binds":[],"configuredReportings":[],"meta":{}}},"appVersion":14,"stackVersion":2,"hwVersion":0,"swBuildId":"1.00.14","zclVersion":3,"interviewCompleted":true,"meta":{"configured":1955752412},"lastSeen":1664521072539,"defaultSendRequestWhen":"immediate"}

Comment (Koenkk) [reply devices/third_reality.js:148]: `"powerSource":"Battery"` looks good so it is not needed. I will remove it.

**Comment by Koenkk** [comment]
Thanks!

File: devices/third_reality.js (hunk 1 above)
Comment (weihuan1111) [reply devices/third_reality.js]: What is the function of the paragraph you deleted？

Comment (danieledwardgeorgehitchcock) [reply devices/third_reality.js]: It sets the correct power source of the device in Z2M if it is not reported by the device itself during interview

**Pull request merged**